tst/com/amazon/ion/benchmark/DataGeneratorTest.java

```diff
@@ -189,13 +198,56 @@ public void testSizeOfGeneratedData() throws Exception {
         assertTrue(difference <= 0.1 * expectedSize);
     }
 
+
+    /**
+     * Test if there's violation by comparing every single data with the Ion schema constraint.
+     * @throws Exception if error occur
+     */
+    @Test
+    public void testViolation() throws Exception {
```

Comment:
It looks like you'll be able to reuse almost all of this test in order to test other `--input-ion-schema` combinations, right? I see that currently it's only testing structs, but you could move most of this into a separate method and reuse it for lists, etc.

Reply:
> It looks like you'll be able to reuse almost all of this test in order to test other `--input-ion-schema` combinations, right? I see that currently it's only testing structs, but you could move most of this into a separate method and reuse it for lists, etc.

Yes, this testViolation method is valid for other `--input-ion-schema` combinations, and the method which tests the violation when generating Ion List has been added.

Reply:
> It looks like you'll be able to reuse almost all of this test in order to test other `--input-ion-schema` combinations, right? I see that currently it's only testing structs, but you could move most of this into a separate method and reuse it for lists, etc.

Yes this method is valid for other combinations as well. In order to avoid duplicated codes, a 'for loop' is added in this method, for each loop, the different ion schema file will be tested.

Reply:
Good, I added a comment related to this on the new revision.